### ISSUE TITLE
Syncing randomly gets stuck in a redirect loop

### ISSUE THREAD

**Issue by osdlge** [octobox/octobox#1020]
I'm running octobox locally under docker-compose. I have the in app auto refresh set to every 30 minutes. 

I am able to reproduce this on the latest build from docker-compose pull (https://github.com/octobox/octobox/commit/11173e998763f2774c99d7f87e9661c1ec48858e)

Are you experiencing this issue in octobox.io or your own instance?
 - [ ] octobox.io
 - [x] My instance

🐞 **Problem**
Occasionally the sync process will get stuck in an infinite redirect loop. This is what it looks like from the browsers dev tools:

![image](https://user-images.githubusercontent.com/12647729/46087583-a1d30e80-c178-11e8-9f2d-01625512885c.png)

The logs will continually repeat that a sync started so a redirect needs to occur

```text
app_1                                | [2018-09-26 14:35:30 +0000] INFO -- <NAME> synced their notifications
app_1                                |
app_1                                |
app_1                                | Redirected to https://<URL>/notifications/sync
app_1                                | Completed 302 Found in 87ms (ActiveRecord: 5.9ms | SQL count: 3)
app_1                                |
app_1                                |
app_1                                | Started GET "/notifications/sync" for <IP> at 2018-09-26 14:35:31 +0000
app_1                                | Cannot render console from <IP>! Allowed networks: 127.0.0.1, ::1, 127.0.0.0/127.255.255.255
app_1                                | Processing by NotificationsController#sync as HTML
app_1                                |   User Load (0.6ms)  SELECT  "users".* FROM "users" WHERE "users"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
app_1                                |   ↳ app/controllers/application_controller.rb:51
app_1                                |   Notification Load (1.3ms)  SELECT  "notifications"."updated_at" FROM "notifications" WHERE "notifications"."user_id" = $1 AND "notifications"."unread" = $2 ORDER BY notifications.updated_at ASC LIMIT $3  [["user_id", 1], ["unread", true], ["LIMIT", 1]]
app_1                                |   ↳ app/services/download_service.rb:56
app_1                                | ETHON: performed EASY effective_url=https://<GITHUB ENTERPRISE URL>/api/v3/notifications?per_page=100 response_code=304 return_code=ok total_time=0.072342
app_1                                |   User Update (3.3ms)  UPDATE "users" SET "last_synced_at" = $1 WHERE "users"."id" = $2  [["last_synced_at", "2018-09-26 14:35:31.090183"], ["id", 1]]
app_1                                |   ↳ app/services/download_service.rb:33
app_1                                |
app_1                                |
app_1                                | [2018-09-26 14:35:31 +0000] INFO -- <NAME> synced their notifications
app_1                                |
app_1                                |
app_1                                | Redirected to <URL>/notifications/sync
app_1                                | Completed 302 Found in 96ms (ActiveRecord: 5.1ms | SQL count: 3)
````
🎯 **Goal**
auto refreshes shouldn't get stuck in an infinite loop

**Comment by andrew** [comment]
I suspect https://github.com/octobox/octobox/pull/1019, which fixed the sync button (https://github.com/octobox/octobox/issues/1017), highlighted a bug with the auto syncing

**Comment by osdlge** [comment]
I was able to reproduce it before that commit, I was hoping that commit would fix it. I update semi-regularly so I can't say for sure when this started :(

**Comment by andrew** [comment]
@osdlge yeah there's been quiet a few changes around the syncing in the past couple weeks, I think part of the cause is this `redirect_back`: https://github.com/octobox/octobox/blob/master/app/controllers/notifications_controller.rb#L242 which is getting `/notifications/sync` as the referrer and starting the loop, I'm going to try adding a check in there to make sure it doesn't redirect back to itself.

**Comment by andrew** [comment]
@osdlge I've just merged #1020, please reopen if this doesn't fix it for you
